### PR TITLE
example/security_test: fix build failure.

### PR DESCRIPTION
**Pull request by Hyunjun85**
Update UTC for security APIs. 

1. fix build failure.
2. Remove malloc without free why It is a memory leak, bug. 
   We have decided that user should pass buffer which is allocated some memory into security API.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f1c8b244c62fdab321fc7119e68a7633ab92782b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f1c8b244c62fdab321fc7119e68a7633ab92782b] - Code Rule Check Result:
apps/examples/security_test/security_api/security_api_ss_test.c:87: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/security_test/security_api/security_api_ss_test.c:104: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [152c2c0b8dc8863f9470c298ed7752296b66ffa6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [152c2c0b8dc8863f9470c298ed7752296b66ffa6] - Code Rule Check OK.

File: apps/examples/security_test/security_api/security_api_ss_test.c [hunk 1]
```diff
@@ -30,9 +30,8 @@
 void
 test_securestorage(void)
 {
-	int i;
+	// int i;
```
Comment (NilabhShant) [inline apps/examples/security_test/security_api/security_api_ss_test.c]: Please remove this line.

File: apps/examples/security_test/security_api/security_api_ss_test.c [hunk 2]
```diff
@@ -76,15 +75,19 @@ test_securestorage(void)
...
+	if (0 != ss_get_list_secure_storage(hnd, &list)) {
 		printf("Fail\n	! ss_get_list_secure_storage\n");
 		goto exit;
 	}
 	printf("ok\n");
 	printf("[%20s] [%8s]\n", "FILE NAME", "FILE ATTR");
-	for (i = 0; i < count; i++) {
-		printf("[%20s] [%08x]\n", list[i].name, list[i].attr);
+	// It will be paased why list will be null always.
+	{
```
Comment (NilabhShant) [inline apps/examples/security_test/security_api/security_api_ss_test.c]: Is this open brace is needed ?

File: apps/examples/security_test/security_api/security_api_ss_test.c [hunk 3]
```diff
@@ -76,15 +75,19 @@ test_securestorage(void)
...
+		security_storage_list ptr = list;
+		while (ptr != NULL) {
+			printf("[%20s] [%08x]\n", ptr->name, ptr->attr);
+			// ptr = ptr->next;
```
Comment (NilabhShant) [inline apps/examples/security_test/security_api/security_api_ss_test.c]: Please remove this line if not needed.

File: apps/examples/security_test/security_api/security_api_ss_test.c [hunk 4]
```diff
@@ -97,8 +100,8 @@ test_securestorage(void)
 
 exit:
 	free_security_data(&output);
-	if (count > 0)
-		free(list);
+	// if (count > 0)
+	// free(list);
```
Comment (NilabhShant) [inline apps/examples/security_test/security_api/security_api_ss_test.c]: same as above.

File: apps/examples/security_test/security_api/security_api_ss_test.c (hunk 1 above)
Comment (Hyunjun85) [reply apps/examples/security_test/security_api/security_api_ss_test.c]: Removed.

**Comment by seinfra** [comment]
Target : [372ab639c2c9fdf2346d52969fb5e93eb301475a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [372ab639c2c9fdf2346d52969fb5e93eb301475a] - Code Rule Check OK.

File: apps/examples/security_test/security_api/security_api_ss_test.c (hunk 2 above)
Comment (Hyunjun85) [reply apps/examples/security_test/security_api/security_api_ss_test.c]: I set checkpoint to implements and I want to share this issue to my partners.
Because the ss_get_list will have to implement.

File: apps/examples/security_test/security_api/security_api_ss_test.c (hunk 3 above)
Comment (Hyunjun85) [reply apps/examples/security_test/security_api/security_api_ss_test.c]: removed.

File: apps/examples/security_test/security_api/security_api_ss_test.c (hunk 4 above)
Comment (Hyunjun85) [reply apps/examples/security_test/security_api/security_api_ss_test.c]: removed.

File: apps/examples/security_test/security_api/security_api_ss_test.c (hunk 2 above)
Comment (NilabhShant) [reply apps/examples/security_test/security_api/security_api_ss_test.c]: Ok.

Comment (Hyunjun85) [reply apps/examples/security_test/security_api/security_api_ss_test.c]: So, the brace will be removed when the function and utc will be updated.

Comment (NilabhShant) [reply apps/examples/security_test/security_api/security_api_ss_test.c]: Ok, Got it.

File: apps/examples/security_test/security_api/security_api_ss_test.c [hunk 5]
```diff
@@ -76,15 +74,18 @@ test_securestorage(void)
...
 		goto exit;
 	}
 	printf("ok\n");
 	printf("[%20s] [%8s]\n", "FILE NAME", "FILE ATTR");
-	for (i = 0; i < count; i++) {
-		printf("[%20s] [%08x]\n", list[i].name, list[i].attr);
+	// It will be paased why list will be null always.
+	{
+		security_storage_list ptr = list;
+		while (ptr != NULL) {
```
Comment (chanijjani) [inline apps/examples/security_test/security_api/security_api_ss_test.c]: It seems that ptr = ptr->next is omitted...

File: apps/examples/security_test/security_api/security_api_ss_test.c [hunk 6]
```diff
@@ -97,8 +98,6 @@ test_securestorage(void)
 
 exit:
 	free_security_data(&output);
-	if (count > 0)
-		free(list);
```
Comment (chanijjani) [inline apps/examples/security_test/security_api/security_api_ss_test.c]: Don't we need to free nodes of the list when the list is not empty?

File: framework/src/security/security_auth.c [hunk 7]
```diff
@@ -235,10 +235,6 @@ security_error auth_get_ecdsa_signature(security_handle hnd, security_ecdsa_para
 		SECAPI_HAL_RETURN(hres);
 	}
 
-	sign->data = (unsigned char *)malloc(h_sign.data_len);
```
Comment (chanijjani) [inline framework/src/security/security_auth.c]: Why did you remove this malloc? Is memory allocated to sign->data inside SECAPI_DATA_DCOPY?

File: framework/src/security/security_keymgr.c [hunk 8]
```diff
@@ -118,22 +118,13 @@ security_error keymgr_get_key(security_handle hnd, security_key_type algo, const
...
 	SECAPI_DATA_DCOPY(h_pubkey, pubkey_x);
 
 	// check pubkey_y for ECDH
 	if (h_pubkey.priv_len == 0) {
 		SECAPI_RETURN(SECURITY_OK);
 	}
 
-	pubkey_y->data = (unsigned char *)malloc(h_pubkey.priv_len);
-	if (!pubkey_y->data) {
-		SECAPI_FREE(pubkey_x);
```
Comment (chanijjani) [inline framework/src/security/security_keymgr.c]: This line is freeing pubkey_x, not pubkey_y so this seems to be correct.

**Comment by Hyunjun85** [comment]
Closed it, the security framework api was updated.
